### PR TITLE
 implement error boundaries

### LINKED ISSUE #367: Set up error boundaries to catch and handle errors gracefully
The program should catch the errors and process them for a more user-friendly experience.
select-error-length

### PR DESCRIPTION
**Pull request by palagdan**
Resolves #367

info-error
info-error1

### CODE REVIEW

**Comment by blcham** [comment]
How hard is it to include that the error comes from processing concrete question/answer?

**Comment by blcham** [comment]
let's discuss it

**Comment by palagdan** [comment]
@blcham 

The issue with my solution is that we should refactor functional components into class components to  catch errors, as functional components unable to handle them in a react. I believe the best approach is to use try-catch blocks wherever possible to log errors or display them within functions. Error boundaries should only be necessary for exceptions that we did not want to catch.

File: src/components/ErrorBoundaries.jsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import React from "react";
+import { Alert, Accordion, Card, Button } from "react-bootstrap";
+
+class ErrorBoundary extends React.Component {
```
Comment (blcham) [inline src/components/ErrorBoundaries.jsx:4]: ErrorBoundaries --> components/ErrorBoundary

File: src/components/ErrorBoundaries.jsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      errorInfo: errorInfo,
+    });
+  }
+
+  render() {
+    if (this.state.hasError) {
+      return (
+        <div style={{ padding: "20px" }}>
+          <Alert variant="danger">
+            <Alert.Heading>Something went wrong.</Alert.Heading>
```
Comment (blcham) [inline src/components/ErrorBoundaries.jsx:30]: It should be localized

File: src/components/ErrorBoundaries.jsx (hunk 1 above)
Comment (palagdan) [reply src/components/ErrorBoundaries.jsx:4]: done

File: src/components/ErrorBoundaries.jsx (hunk 2 above)
Comment (palagdan) [reply src/components/ErrorBoundaries.jsx]: done

**Pull request merged**